NoSuchMethodError: javax.xml.bind.ServiceLoaderUtil.lookupUsingOSGiServiceLoader

When running the AVM Fritz binding in the demo app in Eclipse (as described [here](https://www.openhab.org/docs/developer/ide/eclipse.html)), I'm seeing the following exception:

```
java.lang.NoSuchMethodError: javax.xml.bind.ServiceLoaderUtil.lookupUsingOSGiServiceLoader(Ljava/lang/String;Ljava/util/logging/Logger;)Ljava/lang/Object;
	at javax.xml.bind.ContextFinder.find(ContextFinder.java:413) ~[?:?]
	at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:721) ~[?:?]
	at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:662) ~[?:?]
	at org.openhab.binding.avmfritz.internal.util.JAXBUtils.initJAXBContextDevices(JAXBUtils.java:42) ~[?:?]
	at org.openhab.binding.avmfritz.internal.util.JAXBUtils.<clinit>(JAXBUtils.java:36) ~[?:?]
	at org.openhab.binding.avmfritz.internal.hardware.callbacks.FritzAhaUpdateCallback.execute(FritzAhaUpdateCallback.java:67) ~[?:?]
	at org.openhab.binding.avmfritz.internal.hardware.FritzAhaContentExchange.onComplete(FritzAhaContentExchange.java:66) ~[?:?]
	at org.eclipse.jetty.client.ResponseNotifier.notifyComplete(ResponseNotifier.java:218) ~[?:?]
	at org.eclipse.jetty.client.ResponseNotifier.notifyComplete(ResponseNotifier.java:210) ~[?:?]
	at org.eclipse.jetty.client.HttpReceiver.terminateResponse(HttpReceiver.java:481) ~[?:?]
	at org.eclipse.jetty.client.HttpReceiver.terminateResponse(HttpReceiver.java:461) ~[?:?]
	at org.eclipse.jetty.client.HttpReceiver.responseSuccess(HttpReceiver.java:424) ~[?:?]
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.messageComplete(HttpReceiverOverHTTP.java:365) ~[?:?]
	at org.eclipse.jetty.http.HttpParser.parseFields(HttpParser.java:1180) ~[?:?]
	at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:1553) ~[?:?]
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.parse(HttpReceiverOverHTTP.java:204) ~[?:?]
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.process(HttpReceiverOverHTTP.java:144) ~[?:?]
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.receive(HttpReceiverOverHTTP.java:79) ~[?:?]
	at org.eclipse.jetty.client.http.HttpChannelOverHTTP.receive(HttpChannelOverHTTP.java:131) ~[?:?]
	at org.eclipse.jetty.client.http.HttpConnectionOverHTTP.onFillable(HttpConnectionOverHTTP.java:169) ~[?:?]
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311) ~[?:?]
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105) ~[?:?]
	at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104) ~[?:?]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336) ~[bundleFile:9.4.38.v20210224]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313) ~[bundleFile:9.4.38.v20210224]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171) ~[bundleFile:9.4.38.v20210224]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.produce(EatWhatYouKill.java:135) ~[bundleFile:9.4.38.v20210224]
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:882) [bundleFile:9.4.38.v20210224]
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1036) [bundleFile:9.4.38.v20210224]
	at java.lang.Thread.run(Thread.java:834) [?:?]
```

I'm using the main branch.

I have also tried running the dwdwetter binding, wich also uses JAXB, and could reproduce the error there, so it seems not to be caused by the binding, but by the runtime environment.

I think this may be caused by a known [ServiceMix](https://issues.apache.org/jira/browse/SM-4310) issue, but have not found a way to fix it.

Unfortunately not. With these changes the following exception is thrown:
`09:25:30.659 [ERROR] [ing.avmfritz.internal.util.JAXBUtils] - Exception creating JAXBContext for devices: Implementation of JAXB-API has not been found on module path or classpath.
javax.xml.bind.JAXBException: Implementation of JAXB-API has not been found on module path or classpath.
	at javax.xml.bind.ContextFinder.newInstance(ContextFinder.java:232) ~[bundleFile:2.3.3]
	at javax.xml.bind.ContextFinder.find(ContextFinder.java:375) ~[bundleFile:2.3.3]
	at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:691) ~[bundleFile:2.3.3]
	at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:632) ~[bundleFile:2.3.3]
	at org.openhab.binding.avmfritz.internal.util.JAXBUtils.initJAXBContextDevices(JAXBUtils.java:42) [bundleFile:?]
	at org.openhab.binding.avmfritz.internal.util.JAXBUtils.<clinit>(JAXBUtils.java:36) [bundleFile:?]
	at org.openhab.binding.avmfritz.internal.hardware.callbacks.FritzAhaUpdateCallback.execute(FritzAhaUpdateCallback.java:69) [bundleFile:?]
	at org.openhab.binding.avmfritz.internal.hardware.FritzAhaContentExchange.onComplete(FritzAhaContentExchange.java:66) [bundleFile:?]
	at org.eclipse.jetty.client.ResponseNotifier.notifyComplete(ResponseNotifier.java:218) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.client.ResponseNotifier.notifyComplete(ResponseNotifier.java:210) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.client.HttpReceiver.terminateResponse(HttpReceiver.java:481) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.client.HttpReceiver.terminateResponse(HttpReceiver.java:461) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.client.HttpReceiver.responseSuccess(HttpReceiver.java:424) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.messageComplete(HttpReceiverOverHTTP.java:365) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.http.HttpParser.parseFields(HttpParser.java:1180) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:1553) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.parse(HttpReceiverOverHTTP.java:204) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.process(HttpReceiverOverHTTP.java:144) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.receive(HttpReceiverOverHTTP.java:79) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.client.http.HttpChannelOverHTTP.receive(HttpChannelOverHTTP.java:131) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.client.http.HttpConnectionOverHTTP.onFillable(HttpConnectionOverHTTP.java:169) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:383) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:882) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1036) [bundleFile:9.4.40.v20210413]
	at java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: java.lang.ClassNotFoundException: com.sun.xml.bind.v2.ContextFactory cannot be found by org.eclipse.jetty.util_9.4.40.v20210413
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:519) ~[?:?]
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:171) ~[?:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:522) ~[?:?]
	at javax.xml.bind.ServiceLoaderUtil.nullSafeLoadClass(ServiceLoaderUtil.java:92) ~[bundleFile:2.3.3]
	at javax.xml.bind.ServiceLoaderUtil.safeLoadClass(ServiceLoaderUtil.java:125) ~[bundleFile:2.3.3]
	at javax.xml.bind.ContextFinder.newInstance(ContextFinder.java:230) ~[bundleFile:2.3.3]
	... 31 more
09:25:30.673 [ERROR] [ing.avmfritz.internal.util.JAXBUtils] - Exception creating JAXBContext for templates: Implementation of JAXB-API has not been found on module path or classpath.
javax.xml.bind.JAXBException: Implementation of JAXB-API has not been found on module path or classpath.
	at javax.xml.bind.ContextFinder.newInstance(ContextFinder.java:232) ~[bundleFile:2.3.3]
	at javax.xml.bind.ContextFinder.find(ContextFinder.java:375) ~[bundleFile:2.3.3]
	at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:691) ~[bundleFile:2.3.3]
	at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:632) ~[bundleFile:2.3.3]
	at org.openhab.binding.avmfritz.internal.util.JAXBUtils.initJAXBContextTemplates(JAXBUtils.java:51) [bundleFile:?]
	at org.openhab.binding.avmfritz.internal.util.JAXBUtils.<clinit>(JAXBUtils.java:37) [bundleFile:?]
	at org.openhab.binding.avmfritz.internal.hardware.callbacks.FritzAhaUpdateCallback.execute(FritzAhaUpdateCallback.java:69) [bundleFile:?]
	at org.openhab.binding.avmfritz.internal.hardware.FritzAhaContentExchange.onComplete(FritzAhaContentExchange.java:66) [bundleFile:?]
	at org.eclipse.jetty.client.ResponseNotifier.notifyComplete(ResponseNotifier.java:218) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.client.ResponseNotifier.notifyComplete(ResponseNotifier.java:210) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.client.HttpReceiver.terminateResponse(HttpReceiver.java:481) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.client.HttpReceiver.terminateResponse(HttpReceiver.java:461) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.client.HttpReceiver.responseSuccess(HttpReceiver.java:424) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.messageComplete(HttpReceiverOverHTTP.java:365) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.http.HttpParser.parseFields(HttpParser.java:1180) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:1553) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.parse(HttpReceiverOverHTTP.java:204) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.process(HttpReceiverOverHTTP.java:144) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.receive(HttpReceiverOverHTTP.java:79) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.client.http.HttpChannelOverHTTP.receive(HttpChannelOverHTTP.java:131) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.client.http.HttpConnectionOverHTTP.onFillable(HttpConnectionOverHTTP.java:169) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:383) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:882) [bundleFile:9.4.40.v20210413]
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1036) [bundleFile:9.4.40.v20210413]
	at java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: java.lang.ClassNotFoundException: com.sun.xml.bind.v2.ContextFactory cannot be found by org.eclipse.jetty.util_9.4.40.v20210413
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:519) ~[?:?]
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:171) ~[?:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:522) ~[?:?]
	at javax.xml.bind.ServiceLoaderUtil.nullSafeLoadClass(ServiceLoaderUtil.java:92) ~[bundleFile:2.3.3]
	at javax.xml.bind.ServiceLoaderUtil.safeLoadClass(ServiceLoaderUtil.java:125) ~[bundleFile:2.3.3]
	at javax.xml.bind.ContextFinder.newInstance(ContextFinder.java:230) ~[bundleFile:2.3.3]
	... 31 more
`
But at least the exception now shows in the log.

--- a/launch/app/app.bndrun
+++ b/launch/app/app.bndrun
@@ -71,9 +71,10 @@ feature.openhab-model-runtime-all: \
 -runee: JavaSE-17
 
 -runrequires.ee: \
+	bnd.identity;id='jakarta.xml.bind-api',\
 	bnd.identity;id='org.apache.servicemix.specs.activation-api-1.2.1',\
 	bnd.identity;id='org.apache.servicemix.specs.annotation-api-1.3',\
-	bnd.identity;id='org.apache.servicemix.specs.jaxb-api-2.3'
+	bnd.identity;id='org.glassfish.hk2.osgi-resource-locator'
 
 -runproperties: \
 	jdk.util.zip.disableZip64ExtraFieldValidation=true,\
@@ -140,6 +141,7 @@ feature.openhab-model-runtime-all: \
 	jakarta.annotation-api;version='[2.0.0,2.0.1)',\
 	jakarta.inject.jakarta.inject-api;version='[2.0.0,2.0.1)',\
 	jakarta.validation.jakarta.validation-api;version='[2.0.2,2.0.3)',\
+	jakarta.xml.bind-api;version='[2.3.3,2.3.4)',\
 	jakarta.xml.soap-api;version='[1.4.2,1.4.3)',\
 	jakarta.xml.ws-api;version='[2.3.3,2.3.4)',\
 	javax.measure.unit-api;version='[2.1.2,2.1.3)',\
@@ -169,8 +171,6 @@ feature.openhab-model-runtime-all: \
 	org.apache.felix.webconsole.plugins.ds;version='[2.2.0,2.2.1)',\
 	org.apache.servicemix.specs.activation-api-1.2.1;version='[1.2.1,1.2.2)',\
 	org.apache.servicemix.specs.annotation-api-1.3;version='[1.3.0,1.3.1)',\
-	org.apache.servicemix.specs.jaxb-api-2.3;version='[2.3.0,2.3.1)',\
-	org.apache.servicemix.specs.stax-api-1.0;version='[2.9.0,2.9.1)',\
 	org.apache.ws.xmlschema.core;version='[2.2.5,2.2.6)',\
 	org.bitbucket.b_c.jose4j;version='[0.9.3,0.9.4)',\
 	org.eclipse.emf.common;version='[2.17.0,2.17.1)',\
@@ -206,6 +206,7 @@ feature.openhab-model-runtime-all: \
 	org.eclipse.xtext.xbase;version='[2.29.0,2.29.1)',\
 	org.glassfish.hk2.external.aopalliance-repackaged;version='[2.4.0,2.4.1)',\
 	org.glassfish.hk2.external.javax.inject;version='[2.4.0,2.4.1)',\
+	org.glassfish.hk2.osgi-resource-locator;version='[1.0.3,1.0.4)',\
 	org.objectweb.asm;version='[9.4.0,9.4.1)',\
 	org.objectweb.asm.commons;version='[9.4.0,9.4.1)',\
 	org.objectweb.asm.tree;version='[9.4.0,9.4.1)',\

